tests/virtualization/xen/hotplugging.pm

```diff
@@ -127,6 +127,11 @@ sub run_test {
         }
     }
 
+    #Workaround to drop all live provisions of all vm guests
+    if (get_var('VIRT_AUTOTEST') && (check_var('SYSTEM_ROLE', 'kvm') || check_var('HOST_HYPERVISOR', 'kvm')) && (($sles_running_version eq '12' and $sles_running_sp eq '5') || ($sles_running_version eq '15' and $sles_running_sp eq '1'))) {
+        record_info "Reboot All Guests", "Mis-handling of live and config provisions by other test modules may have negative impact on 12-SP5 and 15-SP1 KVM scenarios due to bsc#1171946. So here is the workaround to drop all live provisions by rebooting all vm guests.";
+        perform_guest_restart;
```

Comment:
Code itself LGTM. I only have one question raising to myself and you too, that is it safe to continue to next whatever test module after 'virsh start' without checking that it is actually up normally? ('virsh start' returns nearly immediately)

Reply:
It also did come up on mind briefly. In fact, virtual machine restart is very safe and quick(I think the most extreme time is less than 10s), it is already up and running again after you type in your next command, let alone these feature tests which have enough post-handling as buffer before proceeding to the next module. As far as "whether it is actually up and running" is concerned, I think he or she can add his or her own check afterwards, because there is no general practice to check whether a virtual machine is really up and running. In fact, as long as "virsh start" returns succesfully, it already indicates successful boot. There is no need to add extra check inside the subroutine, additional check should be customized by someone uses this function. But if you think this not safe enough, I can use assert_script_run instead script_run which is easy and more safe.

Reply:
> It also did come up on my mind briefly.
Glad so.

>As far as "whether it is actually up and running" is concerned, I think he or she can add his or her own check afterwards, 

This is the point!
Firstly, I would suggest to add a comment to highlight this for users. 
Secondly, since the test modules have already been submitted, and this function is introduced later and applied since this PR, you need to be careful about it -- so I am afraid you need to check whether the guest booting state(after virsh start) is the proper state for the following test modules to start with, whether they properly handled the checking or state preparation if needed. And if not, the relevant changes should be also included here to not break following tests.

If the second one is properly considered and handled already, I am glad to merge it if no further comment from others.

>because there is no general practice to check whether a virtual machine is really up and running, any specific checking inside will compromise the independence of this subroutine. In fact, as long as "virsh start" returns succesfully, it already indicates successful boot. There is no need to add extra check inside the subroutine, additional check should be customized by someone uses this function outside. But if you think this is not safe enough, I can use assert_script_run instead of script_run which is easy and more safe.

Reply:
Firstly, comment added
Secondly, everything should be OK for the following test modules